### PR TITLE
Convert ComponentPreview.js into a functional component

### LINKED ISSUE #3502: Convert ComponentPreview.js into a functional component
`src/components/ComponentPreview.js` is a class component when it doesn't need to be.
Convert it to a functional component.

pure component would essentially look like this:
```
const ComponentPreview = (props) => {
  const {componentName, hasReact, etc} = props;
  return (
    <component code here>
  )
}
```

This is done similarly in this file: https://github.com/sparkdesignsystem/spark-design-system/blob/d950981d1165033359aaae04b0fe2e6890c78e86/src/components/DevDocLinks.js
(currently part of an open PR)

### AC:
1. Converted `ComponentPreview` into a functional component
2. Remove useless constructor
3. Make a PR off of `staging` branch

image

### PR DESCRIPTION
**Pull request by vivek32ta**
## What does this PR do?
Converted ComponentPreview into a functional component and removed constructor at `src/components/ComponentPreview.js`

### Associated Issue
Fixes #3502 

### Documentation
 - [ ] Update Spark Docs

### Code
 - [ ] Build Component in HTML
 - [ ] Build Component in Angular
 - [ ] Build Component in React
 - [ ] Unit Testing in HTML with `npm run test` in `html/` (100% coverage, 100% passing)
 - [ ] Unit Testing in Angular with `npm run test` in `angular/` (100% coverage, 100% passing)
 - [ ] Unit Testing in React with `npm run test` in `react/` (100% coverage, 100% passing)

### Browser Testing (current version and 1 prior)
  - [ ] Google Chrome
  - [ ] Google Chrome (Mobile)
  - [ ] Mozilla Firefox
  - [ ] Mozilla Firefox (Mobile)
  - [ ] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [ ] Microsoft Edge
  - [ ] Apple Safari
  - [ ] Apple Safari (Mobile)
  
### Accessibility Testing
  - [ ] Axe browser extension
  - [ ] Jaws Inspect
  - [ ] VoiceOver (iOS) or Talkback (Android)

### Deploy Preview Reviewed and Approved
 - [ ] Design Team
 - [ ] Accessibility Team

### CODE REVIEW

**Review by guern1kn — CHANGES_REQUESTED** [review]
Thanks for the PR, @vivek32ta ! Let's update this to be an arrow function.

**Comment by vivek32ta** [comment]
Hi @guern1kn 

I've updated `ComponentPreview` to an arrow function as you asked. And thanks a lot for letting me contribute.

I'll be looking forward to contributing more.

Best,
Vivek

File: src/components/ComponentPreview.js [hunk 1]
```diff
@@ -3,55 +3,49 @@ import PropTypes from 'prop-types';
...
+    componentName,
+    hasReact,
+    hasAngular,
+    hasHTML,
+    componentType,
+    minHeight,
+    maxHeight,
+    maxWidth,
+    allowScrolling,
+  } = props;
```
Comment (Ahmeed2m) [inline src/components/ComponentPreview.js:17]: I would consider destructing the props directly would be a better approach just as a better practice.

```suggestion
const ComponentPreview = ({
    componentName,
    hasReact,
    hasAngular,
    hasHTML,
    componentType,
    minHeight,
    maxHeight,
    maxWidth,
    allowScrolling,
  }) => {
```

Comment (afebbraro) [reply src/components/ComponentPreview.js:17]: Love this! Less code.

File: src/components/ComponentPreview.js [hunk 2]
```diff
@@ -3,55 +3,48 @@ import PropTypes from 'prop-types';
...
+    componentName,
+    hasReact,
+    hasAngular,
+    hasHTML,
+    componentType,
+    minHeight,
+    maxHeight,
+    maxWidth,
+    allowScrolling,
+  }) => {
```
Comment (guern1kn) [inline src/components/ComponentPreview.js:16]: This section is indented one too many.

**Review by guern1kn — CHANGES_REQUESTED** [review]
Thanks for the updates! There's one section that I left a comment on in regards to spacing. After that, we should be good to get this merged! Also, please update this with the staging branch.

File: src/components/ComponentPreview.js (hunk 2 above)
Comment (vivek32ta) [reply src/components/ComponentPreview.js]: I've made the changes you've asked for. And I updated my staging with upstream too.

**Review by guern1kn — APPROVED** [review]
Thank you for contributing!

**Comment by vivek32ta** [comment]
@guern1kn @Ahmeed2m @afebbraro 

Thanks a lot to y'all for holding my hand through the entire process and helping me with improvements. I truly appreciate it.

More power to you!

Best,
Vivek

**Pull request merged**